custom_components/ttlock/coordinator.py

```diff
@@ -179,6 +197,15 @@ def _process_webhook_data(self, event: WebhookEvent):
                 new_data.last_user = event.user
                 new_data.last_reason = event.event.description
 
+        if sensorState := event.sensorState:
+            if sensorState.opened == SensorState.opened:
+                new_data.opened = True
+            if sensorState.opened == SensorState.closed:
+                new_data.opened = False
+                new_data.locked = True
```

Comment:
Is this always the case? Does this need to check if autolock is configured?

Reply:
Yes, if the lock has a door sensor then it will lock when closed (you cannot change this setting, setting auto lock on/off is irrelevant)